### ISSUE TITLE
[Payment card / Subscription] make declined credit card have better error message

### ISSUE THREAD

**Issue by blimpich** [Expensify/App#49359]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Reproducible in staging?:** yes
**Reproducible in production?:** yes
**Email or phone of affected tester (no customers):**
**Issue reported by:** @anmurali 
**Slack conversation:** https://expensify.slack.com/archives/C036QM0SLJK/p1726609164016559

## Action Performed:
Break down in numbered steps
1. create a new account with a collect subscription
2. go to the subscription page and try to add a billing card using [Stripe's test card that will decline due to insufficient funds](https://docs.stripe.com/testing#declined-payments) (`4000000000009995`)
3. Try to add it

## Expected Result:
The error provided to the user would be well worded and let them know what went wrong and what they can do to fix it

## Actual Result:
It says "Auth CreateFund returned an error"

## Screenshots/Videos

![image](https://github.com/user-attachments/assets/b9b11580-4674-4a58-a8e4-8d676509a53c)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by blimpich** [comment]
Ah, I went and looked at [the design doc](https://docs.google.com/document/d/1jXV_vIsOGZuL0dMiuXDWRETAFRX64ODVAhWPtiuFXT8/edit?pli=1) and this is a error case I think we anticipated, but for some reason it isn't working like we originally designed it. Maybe it never worked to begin with and we're only now discovering it because real users are running into it 🤷

![Screenshot 2024-09-17 at 3 29 29 PM](https://github.com/user-attachments/assets/6598a76d-ab57-4f4e-8005-a3e4b82621d6)

Still investigating

**Comment by blimpich** [comment]
I suspect the backend isn't properly updating the billing status onyx key

**Comment by blimpich** [comment]
Well maybe the original hypothesis was correct. It looks like those special errors screens are for when we bill users who were already able to add valid funds to their subscription, whereas this is about being rejected straight from the get go of trying to add a card. So I think it makes sense that we should just provide a generic-ish error message to the user trying to add a bad billing card.

**Comment by blimpich** [comment]
I think I got the solution for this in my head but I ran out of time today. Will come back to it tomorrow. Basically we want to modify web to show a generic error here instead of the Auth CreateFund error. The only thing that is kind of throwing me for a loop is that I'm getting different errors in the Old Dot version of this even though they both call the same CreateFund command under the hood. Need to look into it a bit more.
